ipn/prefs.go

```diff
@@ -244,6 +244,17 @@ func (p *Prefs) ApplyEdits(m *MaskedPrefs) {
 	}
 }
 
+// IsEmpty reports true when there are no masks set or if m is nil.
+func (m *MaskedPrefs) IsEmpty() bool {
+	if m == nil {
+		return true
+	}
+	m2 := *m
+	m2.Prefs = Prefs{}
+	var empty MaskedPrefs
+	return reflect.DeepEqual(m2, empty)
```

Comment:
I think it'd be cheaper to do this with a reflect loop like Prefs.ApplyEdits instead of using DeepEqual (which allocates here)